src/env/root/mod.rs

```diff
@@ -59,9 +59,28 @@ impl<'a> RootRuntime<'a> {
             call_targets: Default::default(),
             call_stack: Default::default(),
             buffer: Default::default(),
+
+            #[cfg(feature = "debug")]
+            logger: RefCell::new(Box::new(|s: &str| {
+                println!("{}", s);
+            })),
         }))
     }
 
+    #[cfg(feature = "debug")]
+    pub fn set_logger(&mut self, f: Box<dyn Fn(&str)>) {
```

Comment:
Could also do it as `fn set_logger<F: Fn(&str)>(&mut self, f: F)` so that if you change the internal storage (`Box`) you don't have to change the interface. Monomorphization has a cost, but I lean towards pleasant interfaces.

Reply:
:+1: